src/components/AssetsTable/AssetsTable.vue

```diff
@@ -202,11 +227,68 @@ export default {
         return direction === 'asc' ? comparison : -comparison
       })
     },
+
+    setSortType(type, column) {
+      return (
+        this.sortConfig.column === column.label &&
+        this.sortConfig.direction === type
+      )
+    },
+
+    controlDialog() {
+      this.visible = true
+    },
+
+    handleFormData(formData) {
+      this.rows.push(formData)
+      this.updatePagination()
+      this.totalItems++
+    },
+
+    formatRowData(rowInput) {
+      if (typeof rowInput === 'string') {
+        return rowInput
+      }
+      return this.formatDate(rowInput)
+    },
+
+    formatDate(rowInput) {
+      const date = new Date(rowInput)
+      return `${date.getFullYear()}-${this.formatNumber(date.getMonth() + 1)}-${this.formatNumber(date.getDate())}`
+    },
+
+    formatNumber(num) {
+      return num < 10 ? '0' + num : num
+    },
   },
 }
 </script>
 
 <style lang="scss">
+.button-add-item {
+  display: flex;
+  align-items: center;
+  gap: 7px;
+  align-self: flex-end;
+  padding: 10px 18px;
+  max-width: 118px;
+  max-height: 36px;
+  border: 1px solid #004b85;
+  border-radius: 16px;
+  background: #004b85;
+  color: #ffff;
+  font-weight: 600;
+  cursor: pointer;
+
+  .pi.pi-plus {
+    font-size: 10px;
+  }
+
+  &-text {
```

Comment:
not vue related, but I'd suggest to avoid writing such classes, because they can be quite difficult to maintain.

for example, I see 'button-add-item-text' class in the template and want to alter it's styles. In order to find It I have to go to scss file/section and just search it with my eyes. While if you had the whole class in your scss it would be just a matter of CMD+F

using it with BEM is ok though (button-add-item__text) we can just search by the block '__text'. While if it's just -text it's unintuitive.

Don't need to fix that though, just fyi